### PR TITLE
Add List-Unsubscribe to broadcasts and day-of reminders

### PR DESCRIPTION
**Pull request by BRCornelius**
## Summary

Team emails were landing in spam. A mail-tester run on the sending domain scored the message **10/10** with only two advisory flags: no DMARC record, and no `List-Unsubscribe` header. DMARC has been published separately (DNS-side, nothing in this repo). This PR closes the second flag on the two sends that are genuinely list mail — the all-parents broadcast, and the day-of reminder cron that arrived from #47 while this PR was open.

## Key Decisions

**The header is set on two sends, not on every email.** RFC 2369 says `List-Unsubscribe` describes *a list the recipient belongs to*. That is true of the all-parents broadcast and the reminder cron, and false of the rest: an invitation is addressed to someone who is not on the team yet, and the `INDIVIDUAL_PARENT` and `ALL_COACHES` shapes are one-to-one correspondence. Blanketing the header across all mail would quiet a deliverability report by putting a false claim in a header, which is a worse trade than an orange checkmark.

The practical consequence is worth stating plainly: **re-running mail-tester against an invitation will still show that flag orange.** That is the intended outcome, not an incomplete fix.

**The address differs between the two, because the senders differ.** A broadcast has a human sending it, so it reuses that coach — the same address already in `Reply-To`. The cron runs as the system with no sender at all, so the contact has to be *chosen*: `pickUnsubscribeContact` prefers the team's owner over an assistant coach (a request to stop reminding a family is really a roster decision) and breaks ties on `userId`, so two runs of the same day name the same person. It is pure and tested, so which coach gets named is a decision rather than a side effect of query ordering. A team with no staff address sends no header rather than one mailing nowhere — and the reminder still goes.

**The address is validated before it is framed, not trusted.** `User.email` is user input — `z.email()` guards the sign-in and bulk-invite forms and the value is then persisted — so being in the database is not what makes it safe. `listUnsubscribeHeader` tests it against a bare addr-spec pattern, because a CR/LF in a header value is header injection and no comment prevents that. A suspect address yields no header rather than a malformed one, and the email still sends.

**The value is a `mailto:`, deliberately not RFC 8058 one-click.** One-click unsubscribe requires an unauthenticated HTTPS POST endpoint and somewhere to record the suppression. For this app that means a parent can silently remove themselves from the only channel that carries "tonight is cancelled, don't drive to the field" — a real hazard for twenty-five families with no second notification path. Routing the unsubscribe to a human who can ask why is the safer default at this scale. The reasoning is in `AGENTS.md` so revisiting it is a deliberate decision about that failure mode rather than an accident while chasing a checkmark.

## What's in Scope

- `sendEmail` accepts an optional `listUnsubscribe` address, validates it, and emits it as `<mailto:addr?subject=Unsubscribe>`. The key is omitted entirely when absent or rejected — not an empty `headers` object — matching the existing `replyTo` handling. The RFC framing lives in this one place rather than at each call site.
- `sendTeamMessageAction` passes the sending coach's address for `ALL_PARENTS` only.
- `pickUnsubscribeContact` (pure, in `reminders.ts`) chooses one staff address per team; `loadTodaysReminderWork` resolves it once per team alongside the roster read, and the cron passes it through.
- Tests at every level: header framing, omission, and four rejected-address cases in `email.test.ts`; audience scoping in `messages/new/actions.test.ts`; contact selection in `reminders.test.ts`; per-team resolution in `reminder-data.test.ts`; the send payload in the cron's `route.test.ts`.
- `AGENTS.md` gotcha covering both senders and why the addresses differ.

### Out of Scope

- **The DMARC record** — DNS, not code. Already published.
- **RFC 8058 one-click unsubscribe.** Needs a suppression column, a migration, an unauthenticated route handler, recipient filtering, and a resubscribe affordance. That is a feature, and it needs a decision on the missed-cancellation hazard first. Reminders are the strongest future candidate — "stop reminding me about every practice" is reasonable in a way that "stop telling me games are cancelled" is not.
- **A `text/plain` alternative part.** Flagged as a suspect during diagnosis, then ruled out — mail-tester did not raise it and the message scored 10/10 on content.

## Bugs Found and Fixed During Review

- **A comment claimed a safety property the code did not enforce.** The first version of `listUnsubscribeHeader` was annotated "the address comes from a stored `User.email`, never from a form" — misleading, since `User.email` is user input validated at entry and then persisted. Copilot flagged the wording. Rewording alone would have left the safety resting on a comment, so the claim became a check: the address is now tested against a bare addr-spec pattern before interpolation, closing a header-injection path. Four regression cases (CRLF, embedded angle bracket, whitespace, missing at-sign) each assert the header is dropped and the send still succeeds; all four fail against the pre-fix code.

## Known Gaps / Follow-ups

- **Deliverability is not verified by this PR and cannot be from CI.** DMARC plus this header close both mechanical gaps, but the remaining variable is Gmail's sender-reputation model on a new, low-volume domain that sends in bursts. That improves with engagement, not configuration. Enrolling `send.corneliuses.com` in Google Postmaster Tools will show whether reputation actually climbs.
- The header's real-world rendering has not been checked against a live send — only the payload handed to Resend is asserted.
- `loadUnsubscribeContactsByTeamId` reads `Membership`, the one place in the reminder path that does. It is not a recipient lookup — nobody there is mailed — but it is worth knowing about if the reminder recipient rules ever change.

## Test Plan

- [ ] Run `pnpm check` — lint, typecheck, and tests. Passing locally at 1440 tests across 98 files, and green in CI on `74ff924`.
- [ ] Run `pnpm exec next build` — the half `pnpm check` does not cover. Green in CI.
- [ ] Confirm the new assertions are not vacuous. Three mutations, all verified during development:
      (a) delete the `...(payload.unsubscribeEmail ? …)` spread from the cron route → *"points the header at the team's staff contact"* fails;
      (b) replace the comparator body in `pickUnsubscribeContact` with `return 0` → *"prefers the owner over an assistant coach"*, *"breaks a tie on userId"*, and the loader's per-team test all fail;
      (c) widen `BARE_ADDRESS` to match anything → all four rejected-address cases fail.
- [ ] Edge cases checked: the two one-to-one audiences, the no-staff-address team, and every rejected address all assert the key is *absent*, not merely falsy; a team with no staff still sends its reminder.

### CODE REVIEW

File: src/lib/email.ts [hunk 1]
```diff
@@ -26,11 +47,18 @@ function requireEnv(name: string): string | null {
   return value ? value : null;
 }
 
+/// RFC 2369: the value is a URL in angle brackets. The address comes from a
+/// stored `User.email`, never from a form.
+function listUnsubscribeHeader(address: string): Record<string, string> {
```
[the remark this answers is not in the document]
Comment (BRCornelius) [reply src/lib/email.ts]: Correct, and the phrasing has been fixed — but rewording alone would have left the safety resting on a comment, so the claim is now a check. Fixed in 74ff924.

`User.email` is user input, validated by `z.email()` at `signin/actions.ts:9` and `roster/invite/actions.ts:23` and then persisted; being in the database is not what makes it safe. `listUnsubscribeHeader` now tests the address against a bare addr-spec pattern before framing it, and a suspect value yields no header rather than a malformed one. The email still sends either way, matching how the rest of the module degrades, and the address is never logged since it is personal contact detail.

The concrete risk this closes is header injection: a CR/LF in the value would otherwise be interpolated straight into a header. `email.ts` is the single place that frames this header, so it is the place that has to check. Four cases are covered in `email.test.ts` — CRLF injection, an embedded angle bracket, whitespace, and a missing at-sign — each asserting the header is dropped and the send still succeeds. Verified non-vacuous by widening the pattern to match anything, which fails all four.

---
_Generated by [Claude Code](https://claude.ai/code)_
